back-end/src/sciler/handler/handler_helper.go

```diff
@@ -183,16 +238,16 @@ func (handler *Handler) getHints() map[string][]string {
 	return hints
 }
 
-// getEventDescriptions returns map of hints with puzzle name as key and list of hints for that puzzle as value
+// getEventDescriptions returns a map of hints with puzzle name as key and list of hints for that puzzle as value
 func (handler *Handler) getEventDescriptions() map[string]string {
 	events := make(map[string]string)
-	for _, rule := range handler.Config.RuleMap {
+	for _, rule := range handler.getPuzzleRules() {
```

Comment:
Why not use a cached map?

Reply:
because the rulemap includes button events which should not be used here